Source/extensions/privilegedrequest/include/privilegedrequest/PrivilegedRequest.h

```diff
@@ -315,11 +315,12 @@ namespace Core {
         private:
 #ifndef __WINDOWS__
             string UniqueDomainName(const string connector) const {
-                TCHAR binder[connector.length() + 1 + 6 + 1];
-                ::strcpy(binder, connector.c_str());
-                ::strcpy(&binder[connector.length()], _T(".XXXXXX")); 
-                ::mktemp(binder);
-                return(string(binder));
+                char* binder = static_cast<char*>(::alloca(connector.length() + 1 + 6 + 1));
+                
+                snprintf(binder, strlen(binder), "%s.XXXXXX", connector.c_str());
```

Comment:
This is a bit f an expensive solution. Why not move the TCHAR binder[connector.length() + 1 + 6 + 1]; -> TCHAR* binder = static_cast<TCHAR*>(::alloca(connector.length() + 1 + 6 + 1));  and maintin the strcpy..